nrf52-hal-common/src/uarte.rs

```diff
@@ -19,22 +25,37 @@ use crate::gpio::{
     PushPull,
 };
 
+use bbqueue::{BBQueue, Producer, Consumer, typenum::*, GrantR};
+
 // Re-export SVD variants to allow user to directly set values
 pub use crate::target::uarte0::{
     baudrate::BAUDRATEW as Baudrate,
     config::PARITYW as Parity,
 };
 
 pub trait UarteExt: Deref<Target = uarte0::RegisterBlock> + Sized {
-    fn constrain(self, pins: Pins, parity: Parity, baudrate: Baudrate) -> Uarte<Self>;
+    fn constrain(self, nvic: NVIC, pins: Pins, parity: Parity, baudrate: Baudrate) -> Uarte<Self>;
```

Comment:
Won't that be far too limiting, requiring ownership of the whole `NVIC` like that?

I've been thinking for a while that we need a higher-level layer on top of NVIC, where each interrupt is its own ownable/movable thing. It's one of those things I want to prototype, but haven't gotten to yet.

Reply:
I moved this to requiring `&mut NVIC`, only needed during the scope of `::new()`.